src/IntType.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types=1);
+
+namespace Fortuneglobe\Types;
+
+use Fortuneglobe\Types\Exceptions\InvalidIntValueException;
+use Fortuneglobe\Types\Interfaces\RepresentsIntValue;
+
+/**
+ * Class IntType
+ * @package Fortuneglobe\Types
+ */
+class IntType extends AbstractType implements RepresentsIntValue
+{
+	/** @var int */
+	private $value;
+
+	public function __construct( int $id )
+	{
+		$this->value = $id;
+	}
+
+	public function toString() : string
+	{
+		return (string)$this->value;
+	}
+
+	public function toInt() : int
+	{
+		return $this->value;
+	}
+
+	public function jsonSerialize() : int
+	{
+		return $this->value;
+	}
+
+	public static function fromString( string $string ) : RepresentsIntValue
+	{
+		$intValue = (int)$string;
+
+		if ( $string !== (string)$intValue )
```

Comment:
PHP covers the following string values in a conversion with `(int)` which our type check won't

* `'+42'` => `42`
* `'-42'` => `42`
* `'0042'` => `42`
* `'+0042'` => `42`
* `'-0042'` => `-42`

I recommend to use `is_numeric()` ... http://php.net/manual/en/function.is-numeric.php

> Finds whether the given variable is numeric. Numeric strings consist of optional sign, any number of digits, optional decimal part and optional exponential part. Thus +0123.45e6 is a valid numeric value.

So the method should be refactored to

```php
	public static function fromString( string $string ): RepresentsFloatValue
	{
		if ( 0 === preg_match( '#^[+-]?\d+$#', $string )
		{
			throw (new InvalidIntValueException())->withString( $string );
		}

		return new static( (int)$string );
	}
```

And not to forget to update the test to cover these values.

Reply:
Agreed. Fixed in e5290f203cd9625cd2ae0aa57ebb8785bb2ee759
Also added integer boundary check so `((string)PHP_INT_MAX) . '1'` and `((string)PHP_INT_MIN) . '1'` are declined.